Sources/HAP/Server/Device.swift

```diff
@@ -48,9 +50,92 @@ public class Device {
     public let setupCode: String
     let storage: Storage
     let pairings: Pairings
-    public let accessories: [Accessory]
+    public var accessories: [Accessory]
     internal var characteristicEventListeners: [Box<Characteristic>: WeakObjectSet<Server.Connection>]
     public var onIdentify: [(Accessory?) -> Void] = []
+    internal var configuration: Configuration
+    public var onConfigurationChange: [(Device) -> Void] = []
+    public let isBridge: Bool
+
+    // The device maitains a configuration number during its life time, which
+    // persists across restarts of the app.
+    //
+    // HAP Specification 5.4: Current configuration number.
+    //
+    // Must update when an accessory, service, or characteristic is added or
+    // removed on the accessory server.
+    // Accessories must increment the config number after a firmware update.
+    // This must have a range of 1-4294967295 and wrap to 1 when it overflows.
+    // This value must persist across reboots, power cycles, etc.
+    //
+    // HAP Specification 2.6.1: Instance IDs
+    //
+    // Instance IDs are numbers with a range of [1, 18446744073709551615]. These
+    // numbers are used to uniquely identify HAP accessory objects within an HAP
+    // accessory server, or uniquely identify ervices, and characteristics
+    // within an HAP accessory object. The instance ID for each object
+    // must be unique for the lifetime of the server/ client pairing.
+    //
+    // HAP Specification 2.6.1.1: Accessory Instance IDs
+    //
+    // Accessory instance IDs, "aid", are assigned from the same number pool
+    // that is global across entire HAP Accessory Server. For example, if the
+    // first Accessory object has an instance ID of "1" then no other Accessory
+    // object can have an instance ID of "1" within the Accessory Server.
+    //
+    internal struct AIDGenerator: Sequence, IteratorProtocol, Codable {
+        internal var lastAID: InstanceID = 1
+        mutating func next() -> InstanceID? {
+            lastAID = lastAID &+ 1 // Add one and overflow if reach max InstanceID
+            if lastAID < 2 {
+                lastAID = 2
+            }
+            return lastAID
+        }
+    }
+
+    internal struct Configuration: Codable {
+        internal var number: UInt32 = 0
+        internal var aidForAccessorySerialNumber = [String: InstanceID]()
+        private var aidGenerator = AIDGenerator()
+
+        // The next aid - should be checked against existing devices to ensure it is unique
+        internal mutating func nextAID() -> InstanceID {
+            return aidGenerator.next()!
+        }
+
+        // Write the configuration record to storage
+        internal func writeTo(_ storage: Storage) {
```

Comment:
I don't like wrapping this inside `Storage`, and having reading/writing in separate objects. However I'm fine with this for now, as it's likely to change and we can move all storage into `Configuration` and drop `Storage`.

Reply:
Yes I agree, let's do that at some stage.